vaadin-list-mixin.html

```diff
@@ -59,6 +60,7 @@
          * Note: unlike `<vaadin-combo-box>`, this property is read-only,
          * so if you want to provide items by iterating array of data,
          * you have to use `dom-repeat` and place it to the light DOM.
+         * @type {Array<HTMLElement!> | null | undefined}
```

Comment:
Actually I just noticed that elsewhere (at least [here](https://github.com/vaadin/vaadin-overlay/blob/master/src/vaadin-focusables-helper.html#L37)) we have used the non-nullable exclamation mark as a prefix instead of suffix for the type so we should probably be consistent with that also.

Also in JSDoc documentation it's shown as a prefix (though it seems to work as a suffix too which I also saw used in some stackoverflow answer which wanted to avoid confusing it with negation).

So maybe it's better to use `!Array<!HTMLElement>` instead of `Array<HTMLElement!>!` for example.

Here's my updated suggestion:
```suggestion
         * @type {!Array<!Element> | undefined}
```

Reply:
Thanks for such a detailed research!  Also, big thanks for JSDoc link, it's very useful.

Updated based on your latest suggestion.